### PR TITLE
[REVIEW ONLY] JAMES-2578 Fix warnings use Mailet new API: Debatable changes

### PR DESCRIPTION
**Pull request by blackheaven**
See #1883

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[28320288231f0fe77ee1fb13d76b954b637a9c04] [ERROR] COMPILATION ERROR : 
[28320288231f0fe77ee1fb13d76b954b637a9c04] [INFO] -------------------------------------------------------------
[28320288231f0fe77ee1fb13d76b954b637a9c04] [ERROR] /james-project/server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java:[256,52] cannot find symbol
[28320288231f0fe77ee1fb13d76b954b637a9c04]   symbol:   variable MAIL_PRIORITY_ATTRIBUTE_NAME
[28320288231f0fe77ee1fb13d76b954b637a9c04]   location: class org.apache.james.queue.jms.JMSMailQueue
[28320288231f0fe77ee1fb13d76b954b637a9c04] [INFO] 1 error
```

File: server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java [hunk 1]
```diff
@@ -44,13 +44,9 @@
 import static org.apache.james.mailrepository.cassandra.MailRepositoryTable.SENDER;
```
Comment (chibenwa) [inline server/mailrepository/mailrepository-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryMailDAO.java]: I am :+1: with this change, however, I would like to agree on the fact that this is a breaking change.

Upgrade procedure: For each Cassandra repository, before the upgrade
 - Create a file repository
 - Create a mailet processor storing emails in that file mail repository
 - Reprocess data of the Cassandra mail repository to the file one

Then perform the upgrade with an empty Cassandra repository. Once done:
 - Create a mailet processor storing emails back in the Cassandra mail repository
 - Reprocess all emails
 - Delete the temporary file mail repository

Seems painful...

File: upgrade-instructions.md [hunk 2]
```diff
@@ -103,3 +103,20 @@ $ nodetool stop
...
+
+Required: Yes
+
+Concerned products: Any product with a JMS implementation active
+
+#### Upgrade procedure
+
+The upgrade should be done with an empty JMS queue.
+
+To empty the JMS queue, start a James with SMTP, LMTP & JMAP being disabled, and wait for the queue to be empty.
```
Comment (chibenwa) [inline upgrade-instructions.md]: JMS mailqueue is widely used.

This procedure is a bit 'hacky"

Why not allow a user to shutdown a given protocol at runtime? That way the upgrade instruction is simpler and don't require manual conf edition (*2) + reboot...

File: server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java [hunk 3]
```diff
@@ -429,14 +436,15 @@ protected void populateMail(Message message, MailImpl mail) throws JMSException
...
+     * @throws IOException 
      */
-    private void setMailAttribute(Message message, Mail mail, String name) {
+    private void setMailAttribute(Message message, Mail mail, String name) throws IOException {
         // Now cast the property back to Serializable and set it as attribute.
         // See JAMES-1241
         Object attrValue = Throwing.function(message::getObjectProperty).apply(name);
 
         if (attrValue instanceof String) {
-            mail.setAttribute(name, SerializationUtil.deserialize((String) attrValue));
```
[the remark this answers is not in the document]
Comment (mbaechler) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: > Furthermore, queue item being 'stream like' data, we could assume a 3.3.0 user have no more 'prior 3.2.0 items' in this queue

You assume people upgrade to every version, that's a broken assumption IMO.

[the remark this answers is not in the document]
Comment (rouazana) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: > You assume people upgrade to every version, that's a broken assumption IMO.

That could be a required path.

Anyway we need to discuss migration strategy, and it should not be handled in this PR. Maybe for all things requiring a migration we can just suppress deprecation warning for now and handle this in an other task

[the remark this answers is not in the document]
Comment (chibenwa) [reply server/queue/queue-jms/src/main/java/org/apache/james/queue/jms/JMSMailQueue.java]: > Maybe for all things requiring a migration we can just suppress deprecation warning for now and handle this in an other task

I 100% agree with that

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by hoangdat** [comment]
Build started sha1 is merged.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Pull request closed without merging**